app/controllers/application_controller/ci_processing.rb

```diff
@@ -1657,6 +1657,49 @@ def scanhosts
     host_button_operation('scan', 'Analysis')
   end
 
+  # Scan all selected or single displayed image(s)
+  def scanimages
+    assert_privileges("image_scan")
+    showlist = @lastaction == "show_list"
+    images = showlist ? find_checked_items : find_scan_item
+
+    if images.empty?
+      add_flash(_("No %{model} were selected for %{task}") % {:model => ui_lookup(:tables => "container_image"),
+                                                              :task  => "Analysis"}, :error)
+    else
+      process_scan_images(images)
+    end
+
+    showlist ? show_list : show
+    images.count
+  end
+
+  def find_scan_item
+    images = []
+    if params[:id].nil? || ContainerImage.find_by_id(params[:id]).nil?
+      add_flash(_("%s no longer exists") % ui_lookup(:table => "container_image"), :error)
+    else
+      images.push(params[:id])
+    end
+    images
+  end
+
+  def process_scan_images(images)
```

Comment:
Since this is container-specific, shouldn't it be in the `ContainerCommonMixin`?

It looks like the `scanimages` method is still fairly generic.  So, moving this to `ContainerCommonMixin` would allow different `ApplicationController` subclasses to implement this as needed.

However, it also looks like the only current caller for `scanimages` is in `ContainerCommonMixin`.  So, it's arguable that `scanimages` should be moved to `ContainerCommonMixin`, too.